File: mobile-app/lib/models/task_comment.dart
class TaskComment {
  final String id;
  final String date;
  final String author;
  final String text;
  final String imageUrl;
  final String fileName;
  final bool isPendingSync;

  TaskComment({
    required this.id,
    required this.date,
    required this.author,
    required this.text,
    required this.imageUrl,
    required this.fileName,
    this.isPendingSync = false,
  });

  factory TaskComment.fromJson(Map<String, dynamic> json) {
    return TaskComment(
      id: json['id'] ?? '',
      date: json['date'] ?? '',
      author: json['author'] ?? '',
      text: json['text'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      fileName: json['fileName'] ?? '',
      isPendingSync: json['isPendingSync'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'date': date,
      'author': author,
      'text': text,
      'imageUrl': imageUrl,
      'fileName': fileName,
      'isPendingSync': isPendingSync,
    };
  }
}

File: backend/src/main/java/com/ilac/model/TaskComment.java
package com.ilac.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TaskComment {
    private String id;
    private String date;
    private String author;
    private String text;
    private String imageUrl;
    private String fileName;
}
